transport/internal/errors/peers.go

```diff
@@ -94,3 +94,43 @@ type ErrInvalidAgentConversion struct {
 func (e ErrInvalidAgentConversion) Error() string {
 	return fmt.Sprintf("cannot convert agent (%v) to type %s", e.Agent, e.ExpectedType)
 }
+
+// ErrNoPeerToSelect is used when a peerlist doesn't have any peers to return
+type ErrNoPeerToSelect string
+
+func (e ErrNoPeerToSelect) Error() string {
+	return fmt.Sprintf("could not find a peer to select in peerlist %s", string(e))
+}
+
+// ErrPeerAlreadyInList is used when a peerlist attempts to add a peer that is already in the list
+type ErrPeerAlreadyInList struct {
+	Peer transport.Peer
+}
+
+func (e ErrPeerAlreadyInList) Error() string {
+	return fmt.Sprintf("can't add peer (%v) because is already in peerlist", e.Peer)
+}
+
+// ErrPeerNotInList is used when a peerlist attempts to remove a peer that is not in the list
```

Comment:
returned to peer providers if the peerlist is not tracking the peer to remove for a given identifier